File: core-common/src/main/java/com/huyingbao/core/common/dialog/CommonLoadingDialog.java
package com.huyingbao.core.common.dialog;

import android.content.DialogInterface;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.huyingbao.core.arch.action.RxActionManager;
import com.huyingbao.core.base.view.CommonDialog;
import com.huyingbao.core.common.R;
import com.huyingbao.core.common.R2;
import com.huyingbao.core.common.utils.CommonUtils;

import javax.inject.Inject;
import javax.inject.Singleton;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import butterknife.BindView;
import butterknife.OnClick;
import dagger.Lazy;

/**
 * 进度提示,通过依赖注入生成
 * <p>
 * Created by liujunfeng on 2019/1/1.
 */
@Singleton
public class CommonLoadingDialog extends CommonDialog {
    @Inject
    Lazy<RxActionManager> mRxActionManagerLazy;

    @BindView(R2.id.progress_bar_loading)
    ProgressBar mProgressBarLoading;
    @BindView(R2.id.tv_loading_notice)
    TextView mTvLoadingNotice;
    @BindView(R2.id.tv_loading_cancel)
    TextView mTvLoadingCancel;

    @StringRes
    private int mMessageInt;


    @Inject
    public CommonLoadingDialog() {
    }

    @Override
    public int getLayoutId() {
        return R.layout.common_dialog_loading;
    }

    @Override
    public void afterCreate(@Nullable Bundle savedInstanceState) {
        getDialog().setCanceledOnTouchOutside(false);
        if (mMessageInt == 0) {
            return;
        }
        String message = getString(mMessageInt);
        if (!TextUtils.isEmpty(message)) {
            mTvLoadingNotice.setText(message);
        }
    }

    @Override
    public void onDismiss(@NonNull DialogInterface dialog) {
        super.onDismiss(dialog);
        mMessageInt = 0;
    }

    @Override
    public void onCancel(@NonNull DialogInterface dialog) {
        super.onCancel(dialog);
        mMessageInt = 0;
    }

    /**
     * 取消所有注册的订阅Action
     */
    @OnClick(R2.id.tv_loading_cancel)
    public void cancel() {
        mRxActionManagerLazy.get().clear();
        CommonUtils.showShortToast(getContext(), "取消操作！");
        dismiss();
    }

    /**
     * 设置进度弹框显示文字
     *
     * @param messageInt
     */
    public void setMessageInt(@StringRes int messageInt) {
        mMessageInt = messageInt;
    }
}

File: core-common/src/main/java/com/huyingbao/core/common/dialog/CommonLoadingDialog.kt
package com.huyingbao.core.common.dialog

import android.content.DialogInterface
import android.os.Bundle
import android.text.TextUtils
import androidx.annotation.StringRes
import com.huyingbao.core.base.dialog.BaseCommonDialog
import com.huyingbao.core.common.R
import kotlinx.android.synthetic.main.common_dialog_loading.*

/**
 * 进度提示
 *
 * Created by liujunfeng on 2019/1/1.
 */
class CommonLoadingDialog : BaseCommonDialog() {
    var clickListener: CommonLoadingDialogClickListener? = null

    @StringRes
    private var messageInt: Int = 0

    companion object {
        fun newInstance(): CommonLoadingDialog {
            return CommonLoadingDialog()
        }
    }

    override fun getLayoutId(): Int {
        return R.layout.common_dialog_loading
    }

    override fun afterCreate(savedInstanceState: Bundle?) {
        dialog!!.setCanceledOnTouchOutside(false)
        if (messageInt != 0) {
            val message = getString(messageInt)
            if (!TextUtils.isEmpty(message)) {
                tv_loading_notice!!.text = message
            }
        }
        tv_loading_cancel.setOnClickListener {
            clickListener?.onCancel()
            dismiss()
        }
    }

    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)
        messageInt = 0
        clickListener = null
    }

    override fun onCancel(dialog: DialogInterface) {
        super.onCancel(dialog)
        messageInt = 0
        clickListener = null
    }

    /**
     * 设置进度弹框显示文字
     */
    fun setMessageInt(@StringRes messageInt: Int) {
        this.messageInt = messageInt
    }
}

/**
 * 点击回调接口
 */
interface CommonLoadingDialogClickListener {
    /**
     * 点击取消
     */
    fun onCancel()
}
